### PR TITLE
TELCODOCS-1141 RN MGMT-11232 Support platform integration in ZTP

### PR DESCRIPTION
**Pull request by kdrake20**
Fixes: RN for [TELCODOCS-1141](https://issues.redhat.com/browse/TELCODOCS-1141)

For: Version 4.12+ 

Doc Preview [here](https://55895--docspreview.netlify.app/openshift-enterprise/latest/release_notes/ocp-4-12-release-notes.html#supported-platforms-for-managed-clusters)

Signed-off-by: Katie Tothill ktothill@redhat.com

### CODE REVIEW

**Comment by ktothill** [comment]
@jelkosz / @filanov and @trewest, can I please get a review of this release note? Thank you!

**Comment by ktothill** [comment]
@filanov and @trewest, could you please confirm if the supported config referenced RN is for 4.12 GA or a Z-stream? Thank you!

**Comment by ktothill** [comment]
Reached out in [#assisted-installer-forum](https://redhat-internal.slack.com/archives/CUPJTHQ5P/p1676487645252249) to confirm if the RN is for 4.12GA or a Z-stream release.

**Comment by ktothill** [comment]
Confirmed the RN is for 4.12, per [#forum-assisted-installer](https://redhat-internal.slack.com/archives/CUPJTHQ5P/p1676487645252249)

File: release_notes/ocp-4-12-release-notes.adoc [hunk 1]
```diff
@@ -1269,6 +1269,17 @@ Optionally, you can now override the default evaluation intervals for all polici
 
 For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-configuring-pgt-compliance-eval-timeouts_ztp-advanced-policy-config[Configuring policy compliance evaluation timeouts for PolicyGenTemplate CRs].
 
+[id="supported-platforms-for-managed-clusters"]
+==== Supported platforms for managed clusters
+
+The Assisted Installer currently supports the following OpenShift platforms:
+
+* BareMetal
```
Comment (jboxman-rh) [inline release_notes/ocp-4-12-release-notes.adoc:1277]:
```suggestion
* Bare metal
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 2]
```diff
@@ -1269,6 +1269,17 @@ Optionally, you can now override the default evaluation intervals for all polici
...
+* VSphere
```
Comment (jboxman-rh) [inline release_notes/ocp-4-12-release-notes.adoc:1278]:
```suggestion
* vSphere
```

File: release_notes/ocp-4-12-release-notes.adoc [hunk 3]
```diff
@@ -1269,6 +1269,17 @@ Optionally, you can now override the default evaluation intervals for all polici
...
+* None
+
+Single node OpenShift does not support VSphere. For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.adoc#supported-platforms-for-managed-clusters_ztp-deploying-far-edge-clusters-at-scale[Supported platforms for managed clusters].
```
Comment (jboxman-rh) [inline release_notes/ocp-4-12-release-notes.adoc:1281]:
```suggestion
Single node OpenShift does not support vSphere. For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.adoc#supported-platforms-for-managed-clusters_ztp-deploying-far-edge-clusters-at-scale[Supported platforms for managed clusters].
```

**Review by jboxman-rh — COMMENTED** [review]
I've changed the platform names to be consistent with what we've been using elsewhere in the docs, which I think is correct. Thanks!

File: release_notes/ocp-4-12-release-notes.adoc (hunk 1 above)
Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc:1277]: Updated as suggested.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 2 above)
Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc:1278]: Updated as suggested.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 3 above)
Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc:1281]: Updated as suggested.

**Comment by trewest** [comment]
Hey @jboxman-rh, they were actually correct before. If the user supplies what is added now, they will not get expected results:
```
The AgentClusterInstall "spoke-0" is invalid: spec.platformType: Unsupported value: "Bare Metal": supported values: "", "BareMetal", "None", "VSphere"
```

**Comment by jboxman-rh** [comment]
> Hey @jboxman-rh, they were actually correct before. If the user supplies what is added now, they will not get expected results:
> 
> ```
> The AgentClusterInstall "spoke-0" is invalid: spec.platformType: Unsupported value: "Bare Metal": supported values: "", "BareMetal", "None", "VSphere"
> ```

Then these should be in backticks, because they're provided as values to the YAML. Thanks!

**Comment by ktothill** [comment]
@trewest @jboxman-rh I've put the requested YAML values back as requested. Could I please get an updated `LGTM` from both of you and @danielerez (I need SME, QE and peer review to merge)? Thank you!!

**Comment by jboxman-rh** [comment]
@ktothill LGTM

**Comment by ktothill** [comment]
@trewest and @danielerez, I've put the requested YAML values back as requested. Could I please get an updated `LGTM` (I need SME and QE review to merge)? Thank you!!

**Review by snarayan-redhat — COMMENTED** [review]
LGTM!
For future PRs, in case you are requesting for another round of review, kindly remove the `peer-review-done` label before putting the `peer-review-needed` label as it might create confusion for the reviewers.

**Comment by kalexand-rh** [comment]
This needs to go through Change Management. From a DPM perspective, it LGTM.

Please ensure that your uses of plain "OpenShift" are correct. Unless it's referring to SNO or a particular named component, it generally needs to be `{product-title}` instead.

**Comment by ktothill** [comment]
CM email sent 2/16

**Comment by sjhala-ccs** [comment]
@ktothill I am guessing this release notes update only applies to 4.12? Do you need this to be cherry picked to 4.13 as well?

File: release_notes/ocp-4-12-release-notes.adoc [hunk 4]
```diff
@@ -1269,6 +1269,17 @@ Optionally, you can now override the default evaluation intervals for all polici
...
+id=["specifying-the-platform-type-for-managed-clusters"]
+==== Specifying the platform type for managed clusters
+
+The Assisted Installer currently supports the following OpenShift platforms:
+
+* `BareMetal`
+* `VSphere`
+* `None`
+
+Single node OpenShift does not support `VSphere`. For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.adoc#specifying-the-platform-type-for-managed-clusters_ztp-deploying-far-edge-clusters-at-scale[Specifying the platform type for managed clusters].
```
Comment (romfreiman) [inline release_notes/ocp-4-12-release-notes.adoc:1281]: vsphere platform. Neither BareMetal. but not sure we should talk in those terms here -  do we have none/baremetal as part of the ztp api?

**Comment by filanov** [comment]
LGTM

**Comment by ktothill** [comment]
@sjhala-ccs, yes this can be cherry picked to 4.13. Thanks!

File: release_notes/ocp-4-12-release-notes.adoc (hunk 4 above)
Comment (danmacpherson) [reply release_notes/ocp-4-12-release-notes.adoc:1281]: @imiller0 -- Does the GitOps ZTP plugin support different platforms? Or is this something we shouldn't document as a part of the network far edge documentation?

File: release_notes/ocp-4-12-release-notes.adoc [hunk 5]
```diff
@@ -1269,6 +1269,17 @@ Optionally, you can now override the default evaluation intervals for all polici
 
 For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-configuring-pgt-compliance-eval-timeouts_ztp-advanced-policy-config[Configuring policy compliance evaluation timeouts for PolicyGenTemplate CRs].
 
+id=["specifying-the-platform-type-for-managed-clusters"]
+==== Specifying the platform type for managed clusters
+
+The Assisted Installer currently supports the following OpenShift platforms:
```
Comment (stevsmit) [inline release_notes/ocp-4-12-release-notes.adoc:1275]: Is there a reason you've spelled out "OpenShift" here and not used {product-title}?

File: release_notes/ocp-4-12-release-notes.adoc (hunk 4 above)
Comment (stevsmit) [inline release_notes/ocp-4-12-release-notes.adoc:1281]: I'm fairly sure there's an attribute for "Single node OpenShift." I think it's {sno-caps}

**Comment by stevsmit** [comment]
@ktothill a few nits

File: release_notes/ocp-4-12-release-notes.adoc (hunk 5 above)
Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc:1275]: Updated as suggested.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 4 above)
Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc]: Updated as suggested.

**Comment by ktothill** [comment]
Awaiting response to Dan's [question](https://github.com/openshift/openshift-docs/pull/55895#discussion_r1112126732) to Ian. Unless Ian suggests changes, RN will be ready to merge.

File: release_notes/ocp-4-12-release-notes.adoc (hunk 4 above)
Comment (imiller0) [reply release_notes/ocp-4-12-release-notes.adoc]: The GitOps flow (using SiteConfig) does not directly expose the platformType. I don't think there is a need to reference the far-edge section here. 

Separately, the section being pointed to does not seem to fit in with the far-edge documentation either. We should review that for purpose/flow and consider removing it. I'll open a separate docs story for that.

Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc]: @danmacpherson, just confirming that your question has been answered and any suggested changes?

Comment (danmacpherson) [reply release_notes/ocp-4-12-release-notes.adoc]: Yep, we'll need to remove following:

"For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-deploying-far-edge-clusters-at-scale.adoc#specifying-the-platform-type-for-managed-clusters_ztp-deploying-far-edge-clusters-at-scale[Specifying the platform type for managed clusters]." 

We'll also need to find a new section for the associated documentation, or possibly remove the content temporarily (as per @imiller0's doc story)

Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc]: Updated as suggested.

**Comment by ktothill** [comment]
Removed content as requested by Dan/Ian.

File: release_notes/ocp-4-12-release-notes.adoc [hunk 6]
```diff
@@ -1269,6 +1269,17 @@ Optionally, you can now override the default evaluation intervals for all polici
 
 For more information, see xref:../scalability_and_performance/ztp_far_edge/ztp-advanced-policy-config.adoc#ztp-configuring-pgt-compliance-eval-timeouts_ztp-advanced-policy-config[Configuring policy compliance evaluation timeouts for PolicyGenTemplate CRs].
 
+id=["specifying-the-platform-type-for-managed-clusters"]
```
Comment (stevsmit) [inline release_notes/ocp-4-12-release-notes.adoc:1272]: ID should be within brackets. This is rendering incorrectly in the preview

Comment (ktothill) [reply release_notes/ocp-4-12-release-notes.adoc]: Updated as suggested.

**Pull request merged**